### PR TITLE
mantlui

### PR DESCRIPTION
**Pull request by ryane**
Mantlui consolidates the web UIs of various components in Mantl, including Mesos, Marathon, Chronos, and Consul at a single url on port 80 (http) or 443 (https) on control nodes.

- [x] Mesos UI (/mesos)
    - [x] Fixes mesos leader redirection (don't have to go to the ui on the leader now)
    - [x] Stream mesos logs
    - [x] View mesos task sandbox
    - [x] Stream mesos task stderr/stdout (and other artifacts)
    - [x] View/download mesos task artifacts
- [x] Marathon UI (/marathon)
- [x] Chronos UI (/chronos)
- [x] Consul UI (/consul)

The existing APIs and UIs are still available at their respective endpoints (mesos at :5050, marathon at :8080, chronos at :4400, and consul at :8500).

### CODE REVIEW

**Comment by stevendborrelli** [comment]
Looking forward to this.

**Comment by siddharthist** [comment]
@ryane First of all, this is awesome. I've tested it on GCE and it works great! However, I'm having some trouble accessing the UIs on DigitalOcean. I can access them on their ports as usual, but as you can see, the UI doesn't work. 
```
~ ➤ curl 45.55.135.102:80 
curl: (7) Failed to connect to 45.55.135.102 port 80: Connection refused
~ ➤ curl 45.55.135.102:443
curl: (7) Failed to connect to 45.55.135.102 port 443: Connection refused
~ ➤ curl 45.55.135.102:8500
<html>
...
</html>
~ ➤ curl 45.55.135.102/consul 
curl: (7) Failed to connect to 45.55.135.102 port 80: Connection refused
```

**Comment by ryane** [comment]
hm, not really sure what the problem is. I built a cluster on DO this morning and was able to access the UIs. 

```
curl -sku admin:$MANTLPW https://resching-do-control-01
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
...
```

Just some ideas of things to double check:

- is the mantlui role enabled in your terraform.yml?
- if so, is the nginx-mantlui service running on control nodes?
- if so, maybe check the logs to see if there were any errors that were preventing nginx from starting.

**Comment by siddharthist** [comment]
@ryane I'll bet it was the `terraform.yml`. Let me try again!

**Comment by siddharthist** [comment]
@ryane You were correct! However, I'm now seeing this error in the Consul UI page (/consul):
```
HTTP error code from Consul: 0 error

This is an error page for the Consul web UI. You may have visited a URL that is loading an unknown resource, so you can try going back to the root.

Otherwise, please report any unexpected issues on the GitHub page.
```

**Comment by ryane** [comment]
did you run `security-setup` to generate a new `security.yml`? It might be that there is a mismatch with mantlui not requiring ssl/auth while consul does.

**Comment by siddharthist** [comment]
Ah damn, that might be it as well. Sorry, I'll try again!

**Comment by siddharthist** [comment]
@ryane You were correct again. After fixing that, when I rebooted all my nodes on DO, the /chronos UI returned a 404. I'll test to see if I can reproduce the issue.

**Comment by stevendborrelli** [comment]
@siddharthist @ryane what is our feeling on this? Are we ready to merge?

**Comment by ryane** [comment]
I think we're close. I haven't been able to repro the issues with consul on digital ocean. @siddharthist what do you think? are you still seeing issues?

**Comment by siddharthist** [comment]
If @ryane wasn't able to reproduce the reboot issue, I'd say go for it!
